lib/trace-writer.js

```diff
@@ -120,14 +120,10 @@ TraceWriter.prototype.flushBuffer_ = function(projectId) {
   // Privatize and clear the buffer.
   var buffer = this.buffer_;
   this.buffer_ = [];
+  var joined = buffer.join();
+  buffer = '[' + joined.slice(0, joined.length -1) + ']';
```

Comment:
Since a bug escaped our testing, we should add unit tests for this:
* Check invariant that buffer_ is an array of stringified objects.
* Check that we are able to `JSON.parse` back the buffer produced by `flushBuffer_`.